packages/common/src/store/premium-content/sagas.ts

```diff
@@ -191,15 +206,132 @@ function* updateNewPremiumContentSignatures({
   }
 }
 
+// Poll premium content signatures for tracks that should have them but do not yet.
+function* handleSpecialAccessTrackSubscriptions(tracks: Track[]) {
+  const currentUserId = yield* select(getUserId)
+  if (!currentUserId) return
+
+  const followeeIds = yield* select(getFolloweeIds)
+  const tippedUserIds = yield* select(getTippedUserIds)
+
+  const statusMap: { [id: ID]: PremiumTrackStatus } = {}
+  const trackParamsMap: { [id: ID]: TrackRouteParams } = {}
+
+  const tracksThatNeedSignature = Object.values(tracks).filter((track) => {
+    const {
+      track_id: trackId,
+      owner_id: ownerId,
+      premium_conditions: premiumConditions,
+      premium_content_signature: premiumContentSignature,
+      permalink
+    } = track
+
+    // Ignore premium content signature only updates, i.e.
+    // make sure the above properties exist before proceeding.
+    if (!trackId || !ownerId || !premiumConditions || !permalink) {
+      return false
+    }
+
+    const hasNoSignature = !premiumContentSignature
+    const isFollowGated = !!premiumConditions?.follow_user_id
+    const isTipGated = !!premiumConditions?.tip_user_id
+    const shouldHaveSignature =
+      (isFollowGated && followeeIds.includes(ownerId)) ||
+      (isTipGated && tippedUserIds.includes(ownerId))
+
+    if (hasNoSignature && shouldHaveSignature) {
+      statusMap[trackId] = 'UNLOCKING'
+      trackParamsMap[trackId] = parseTrackRouteFromPermalink(permalink)
+      return true
+    }
+    return false
+  })
+
+  yield* put(updatePremiumTrackStatuses(statusMap))
+
+  const remoteConfigInstance = yield* getContext('remoteConfigInstance')
+  yield* call(remoteConfigInstance.waitForRemoteConfig)
+  const gatedTrackPollIntervalMs = remoteConfigInstance.getRemoteVar(
+    IntKeys.GATED_TRACK_POLL_INTERVAL_MS
+  )
+
+  yield* all(
+    tracksThatNeedSignature.map((track) => {
+      const trackId = track.track_id
+      return call(pollPremiumTrack, {
```

Comment:
Users might have a lot of gated tracks so this might lead to a lot of requests. I fear we might hit the 6 requests to the same domain limit once profiles have a ton of gated tracks. It feels like we could be smart somehow with the confirmer on follow/tip and fetch _once_ for signatures per track? Or is the signature not guaranteed to be there after indexing?

EDIT: I think my idea might miss the point of this PR though. Would this work?
1. User loads a profile, immediately clicks follow
2. Tracks start loading in from that first request, without signatures since the follow isn't indexed
3. Follow gets indexed, confirmer dispatches an action, triggering a refetch for tracks with follower premium conditions _once_ (this could be the same for the normal flow of unlocking too)

It just _feels_ like there shouldn't be a need to poll separately here but I don't know enough to know for sure

Reply:
good point. i think your confirmer logic here would work for follow-gated tracks but not tip-gated tracks as tipping does not use the confirmer to determine that a tip has been sent.

probably still a good idea for follow-gated tracks to use the confirmer i.e. instead of adding the followeeId right after clicking follow, wait until confirmer confirms the follow. but that means the UI wouldn't show 'UNLOCKING' in this case until confirmer confirms the follow... which is not exactly the UI spec.

also, this whole PR is to handle an edge case. there is only a relatively small window of time when this can happen. maybe we should just ignore the 6 requests limit scenario when users have very many follow-gated tracks, and say that we're okay with some of those requests getting blocked for a second or so? thoughts?